### DISCUSSION TITLE
How to Capturing Telemetry Event from V1 Player ?

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Knowlg/Community#233] [Q&A] [answered]
Hi,

I wanted to know how we can capture the telemetry event generated by the V1 content-player.

For the V2 Player (such as the latest PDF player), we can capture it in the following way:
`
playerElement?.addEventListener("telemetryEvent", handleTelemetryEvent);
`

However, I am having trouble figuring out how to capture the telemetry event from the V1 Player. I have tried the following methods, but unfortunately, none of them seem to work:

1. `
playerElement.contentWindow.addEventListener("renderer:telemetry:event", (event: any) => {
    console.log("V1 player tele event", event);
});
`

2. `
(window as any).frames.contentPlayer.addEventListener("renderer:telemetry:event", (event: any) => {
    console.log("V1 player telemetry event", event);
});
`
3. `playerElement.addEventListener("renderer:telemetry:event", (event: any) => {
    console.log("V1 player telemetry event", event);
});
`

We are trying to track the process of H5P, HTML, and YouTube content in our application where we are using the V1 Player. 
If anyone has insights into capturing telemetry events from the V1 Player, I would really appreciate your help!

@itsvick @suvarnakale @rushi-tekdi

@pallakartheekreddy @HarishGangula

**Comment by HarishGangula** [comment]
@rajnishdargan  Refer https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/ab142f7e45cacd39b7a6f87807971a56fa238994/src/app/client/src/app/modules/player-helper/components/player/player.component.ts#L210 for integration and to add the listener.

**Comment by rajnishdargan** [comment]
Hi @HarishGangula 
The following code works: [Reference](https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/release-7.0.0/src/app/client/src/app/modules/player-helper/components/player/player.component.ts#L188)

```
preview.addEventListener("renderer:telemetry:event", (event: any) => {
  console.log("V1 player telemetry event ===>", event);

  if (event.detail.telemetryData.eid === 'START') {
    console.log("V1 player telemetry START event ===>", event);
  }
  if (event.detail.telemetryData.eid === 'END') {
    console.log("V1 player telemetry END event ===>", event);
  }
});
```
The telemetry events are captured only if I include the `sunbird-plugins/renderer/org.sunbird.iframeEvent-1.0` folder in my parent application(In our case it's Next.js app so we are keeping the sunbird-plugins folder in public folder): [sunbird-plugins](https://github.com/Sunbird-Ed/SunbirdEd-portal/tree/release-7.0.0/src/app/sunbird-plugins/renderer/org.sunbird.iframeEvent-1.0).

Additionally, I’ve configured the playerConfig as follows:
```
playerConfig = {
  config: {
    ...
    repos: ["/sunbird-plugins/renderer"],
    plugins: [
      {
        id: "org.sunbird.iframeEvent",
        ver: 1,
        type: "plugin",
      },
    ],
    ...
  },
  context: {
    ...
  }
};
```
Does this approach look correct for capturing telemetry events from the V1 Player?
Or do you recommend a better solution?

@itsvick @suvarnakale @rushi-tekdi
@pallakartheekreddy

**Reply by HarishGangula** [reply, marked as answer]
This implementation is fine

**Reply by rajnishdargan** [reply]
Okay, Thanks!
